### PR TITLE
envoy: Move to Envoy API v3

### PR DESCRIPTION
**Pull request by jrajahalme**
Update Envoy API from v2 to v3.

This PR does not update the Istio sidecar image in order to verify compatibility with Envoy using v2 APIs. The API features we use should be on-the-wire compatible between these versions.

### CODE REVIEW

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-gke

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-gke

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-gke

**Comment by jrajahalme** [comment]
test-gke

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-gke

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-gke

**Review by aanm — APPROVED** [review]

**Comment by jrajahalme** [comment]
Current GKE build link: https://jenkins.cilium.io/job/Cilium-PR-K8s-GKE/1758/

File: pkg/envoy/xds/server.go [hunk 1]
```diff
@@ -321,7 +321,7 @@ func (s *Server) processRequestStream(ctx context.Context, streamLog *logrus.Ent
 			state := &typeStates[index]
 			watcher := s.watchers[typeURL]
 
-			nodeIP, err := IstioNodeToIP(req.GetNode())
+			nodeIP, err := IstioNodeToIP(req.GetNode().GetId())
```
Comment (aditighag) [inline pkg/envoy/xds/server.go:324]: IstioNodeToIP is still returning an 'Ip'?

File: pkg/envoy/sort_test.go [hunk 2]
```diff
@@ -20,45 +20,55 @@ import (
...
 var _ = Suite(&SortSuite{})
 
-var HeaderMatcher1 = &envoy_api_v2_route.HeaderMatcher{
-	Name:                 "aaa",
-	HeaderMatchSpecifier: &envoy_api_v2_route.HeaderMatcher_RegexMatch{RegexMatch: "aaa"},
+var HeaderMatcher1 = &envoy_config_route.HeaderMatcher{
+	Name: "aaa",
+	HeaderMatchSpecifier: &envoy_config_route.HeaderMatcher_SafeRegexMatch{SafeRegexMatch: &envoy_type_matcher.RegexMatcher{
+		EngineType: googleRe2, Regex: "aaa"},
+	},
```
Comment (aditighag) [inline pkg/envoy/sort_test.go:38]: nit: alignment
Please check changes below as well.

**Comment by jrajahalme** [comment]
This needs an update to the merged `cilium/proxy` SHAs.

File: pkg/envoy/xds/server.go (hunk 1 above)
Comment (jrajahalme) [reply pkg/envoy/xds/server.go:324]: Yes, the return value did not change here; it is the IP address that is by (Istio) convention embedded into the xDS Node Id.

File: pkg/envoy/sort_test.go (hunk 2 above)
Comment (jrajahalme) [reply pkg/envoy/sort_test.go:38]: Alignment is OK as long as it passes `go vet` which is run by `make`.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-gke

**Review by aditighag — APPROVED** [review]

**Comment by jrajahalme** [comment]
netnext hit a known flake #11847

**Pull request merged**